modules/telemetry/examples/log_sink.cpp

```diff
@@ -0,0 +1,22 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+#include "hephaestus/telemetry/absl_log_sink.h"
+#include "hephaestus/telemetry/struclog.h"
+#include "hephaestus/utils/stack_trace.h"
+
+namespace ht = heph::telemetry;
+// NOLINTBEGIN google-build-using-namespace
+using namespace heph::telemetry::literals;
+// NOLINTEND google-build-using-namespace
+
+// NOLINTNEXTLINE(bugprone-exception-escape)
+auto main(int /*argc*/, const char* /*argv*/[]) -> int {
+  const heph::utils::StackTrace stack;
+
+  ht::registerLogSink(std::make_unique<ht::AbslLogSink>());
+
+  const std::string a = "testing absl log sink";
+  const int num = 123;
+  ht::log(ht::LogEntry{ ht::Level::Warn, a } | "num"_f(num));
```

Comment:
This is great!! Love the literals, I guess this works also like?
```suggestion
  ht::log(ht::LogEntry{ ht::Level::Warn, a } | {"num", num});
```
Ideally it would be even more awesome if this would work:

```suggestion
  ht::log({ ht::Level::Warn, a } | "num"_f(num));
```

but I guess C++ is not able to deduce the type?

Reply:
also it would be interesting to start thinking how we could handle `log_if` and `log_every_n` or `log_every_ts` .

Reply:
added an ticket for this here https://github.com/olympus-robotics/hephaestus/issues/199 to handle in a seperate PR

Reply:
about 
´´´cpp
ht::log({ ht::Level::Warn, a } | "num"_f(num));
```
I have to try around with that, ht::log has only one single defnition as log(const LogEntry&) so it should work as 
´´´cpp
ht::log({ ht::Level::Warn, a });
```
not sure about the piping though